src/arch/Sound/RageSoundDriver_WDMKS.cpp

```diff
@@ -1072,15 +1073,17 @@ namespace
 		{
 		case DeviceSampleFormat_Float32: // untested
 		{
-			float *pOutBuf = (float *) pOut;
+			auto* pOutBuf = static_cast<float*>(pOut);
 			for( int i = 0; i < iSamples; ++i )
+			{
 				pOutBuf[i] = SCALE( pIn[i], -32768, +32767, -1.0f, +1.0f ); // [-32768, 32767] -> [-1,+1]
+			}
 			break;
 		}
 		case DeviceSampleFormat_Int24:
 		{
-			const unsigned char *pInBytes = (unsigned char *) pIn;
-			unsigned char *pOutBuf = (unsigned char *) pOut;
+			const auto* pInBytes = reinterpret_cast<const unsigned char*>(pIn);

```

Comment:
Why reinterpret_cast here?

Reply:
This only worked as is due to the nature of C style casts. This is needed to update the cast to C++ style. 

If you try to do 

`const unsigned char* pInBytes = static_cast<const unsigned char*>(pIn);`

you will get

```
Error	C2440	'static_cast': cannot convert from 'const int16_t *' to 'const unsigned char *'
				ITGmania		RageSoundDriver_WDMKS.cpp	1082		
```